### PR TITLE
Improve docs in README

### LINKED ISSUE #7: This library in comparison to
How does this library differ from these libraries...?

[android-priority-jobqueue](https://github.com/yigit/android-priority-jobqueue)
[android-job](https://github.com/evernote/android-job)

thank you

### PR DESCRIPTION
**Pull request by ciarand**
Removes a bunch of the "concepts" stuff as it wasn't particularly
helpful. Added an "Overview" and a "Quick start" section, as well as
links to the memory and battery optimizations I/O talk.

Resolves #7, helps with #11 and #13

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,202 +3,212 @@
...
+Bundle myExtrasBundle = new Bundle();
+myExtrasBundle.putString("some_key", "some_value");
+
+Job myJob = dispatcher.newJobBuilder()
+    .setService(MyJobService.class)                      // the JobService that will be called
+    .setTag("my-unique-tag")                             // uniquely identifies the job
+    .setRecurring(false)                                 // one-off job
+    .setLifetime(Lifetime.UNTIL_NEXT_BOOT)               // don't persist past a device reboot
+    .setTrigger(Trigger.executionWindow(0, 60))          // start between 0 and 60 seconds from now
+    .setReplaceCurrent(false)                            // don't overwrite an existing job
```
Comment (morepork) [inline README.md]: existing job + with the same tag

File: README.md [hunk 2]
```diff
@@ -3,202 +3,212 @@
...
-### Job
-
-A Job is a description of a unit of work. At its heart, it's composed of a
-series of mandatory attributes:
+The JobScheduler is an Android system service available on API levels 21 (Lollipop)+. It provides an
+API for scheduling units of work (represented by [`JobService`][JobService] subclasses) that will be
+executed in your app's process.
 
--   A string tag that (within your app) uniquely identifies the job.
+### Why is this better than whatever I'm doing now?
```
Comment (ogeidix) [inline README.md]: s / whatever I'm doing now / background services and listening for system broadcast

File: README.md [hunk 3]
```diff
@@ -3,202 +3,212 @@
 [ci-badge]: https://travis-ci.org/firebase/firebase-jobdispatcher-android.svg?branch=master
 [ci-link]: https://travis-ci.org/firebase/firebase-jobdispatcher-android
 
-The Firebase Android JobDispatcher is a library that provides a high-level wrapper around job
-scheduling engines on Android, starting with the [GCM Network Manager][nts].
+The Firebase JobDispatcher is a library for working with the [JobScheduler][] compatibility library
```
Comment (ogeidix) [inline README.md]: s/ for working with / to schedule future and periodic background jobs in your app. It uses

Comment (ogeidix) [inline README.md]: s/ the [JobScheduler][] compatibility library ... / This library uses the services provided by the Google Play services application to extend the [Android JobScheduler APIs] to android versions before Lollipop.

**Comment by ogeidix** [comment]
Added few minor suggestions.
Feel free to ignore them if the original version looks better to you.

File: README.md [hunk 4]
```diff
@@ -3,202 +3,212 @@
...
+### Comparison to other libraries
 
-`Driver` is an interface that represents a component that can schedule, cancel,
-and execute Jobs. The only bundled Driver is the `GooglePlayDriver`, which
-relies on the scheduler built-in to Google Play services.
+| Library                    | Minimum API | Requires Google Play   | Service API<sup>[1](#fn1)</sup> | Custom retry strategies |
+| -------------------------- | ----------- | ---------------------- | ------------------------------- | ----------------------- |
+| Framework [JobScheduler][] | 21          | No                     | JobScheduler                    | Yes                     |
+| Firebase JobDispatcher     | 9           | Yes                    | JobScheduler                    | Yes                     |
+| [evernote/android-job][]   | 9 / 14      | No<sup>[2](#fn2)</sup> | Custom                          | Yes<sup>[3](#fn3)</sup> |
```
Comment (vRallev) [inline README.md]: Minimum API 14.

File: README.md [hunk 5]
```diff
@@ -3,202 +3,212 @@
...
 
-### Trigger
+<a name="fn1">1</a>: Refers to the methods that need to be implemented in the Service subclass.  
+<a name="fn2">2</a>: Uses AlarmManager to support API levels >= 14 if Google Play services is unavailable
```
Comment (vRallev) [inline README.md]: That's wrong, it should be "Uses AlarmManager to support API levels < 21 if Google Play services is unavailable.

**Comment by ciarand** [comment]
Hey @vRallev, thanks for your comments! Can you have a look at the current README (post #38) and let me know if I've got anything wrong re: the evernote library?

**Comment by vRallev** [comment]
@ciarand Done, I've uploaded #41

**Pull request closed without merging**